src/components/ebay-avatar/index.marko

```diff
@@ -0,0 +1,39 @@
+import { processHtmlAttributes } from "../../common/html-attributes";
+import {getColorForText} from './util';
+
+static var ignoredAttributes = [
+    "a11yText",
+    "class",
+    "size",
+    "color",
+    "img",
+    "username"
+];
+
+$ var isSignedOut = !input.username;
+$ var color = getColorForText(input.username, input.color)
+
+<div
+    ...processHtmlAttributes(input, ignoredAttributes)
+    role="img"
+    aria-label=input.a11yText
+    class=[
+        "avatar",
+        input.class,
+        input.size && `avatar--${input.size}`,
+        `avatar--${color}`,
+    ]>
+    <if(isSignedOut) >
+        <ebay-avatar-signed-out-icon/>
+    </if>
+    <else-if(input.img)>
+        <img ...input.img/>
+    </else-if>
+    <else-if(input.username)>
+
+        ${input.username.charAt(0).toUpperCase()}
+    </else-if>
+    <else>
+        <${input.renderBody}/>
```

Comment:
This branch will never be reached, because `isSignedOut` will be true every time `input.username` is falsy and vice versa.